product-service/controllers/products.controller.ts

```diff
@@ -1,23 +1,69 @@
+import { Client } from 'pg';
+
 // models
 import { ProductModel } from '../models/product.model';
 
-// db
-import { MOCK_DB } from '../mocks/db';
-
 export const getAllProducts = async (): Promise<ProductModel[]> => {
-    return new Promise((resolve) => {
-        setTimeout(() => {
-            resolve(MOCK_DB.products);
-        }, 2000);
-    });
+    const dbConnection = new Client();
+
+    try {
+        await dbConnection.connect();
+
+        const { rows: products } = await dbConnection.query(`
+            select id, title, description, price, photo_url, stocks.count from products
+            inner join stocks
+            on products.id = stocks.product_id
+        `);
+
+        return products;
+    } catch (error) {
+        console.log(error);
+    } finally {
+        dbConnection.end();
+    }
+};
+
+export const getProductById = async (productId: string): Promise<ProductModel> => {
+    const dbConnection = new Client();
+
+    try {
+        await dbConnection.connect();
+
+        const { rows: products } = await dbConnection.query(`
+            select id, title, description, price, photo_url, stocks.count from products
+            inner join stocks
+            on products.id = stocks.product_id
+            where id = '${productId}'
```

Comment:
It's better to use this:
https://node-postgres.com/features/queries

Reply:
Fixed